File: spring-boot-sampler/policies/src/main/java/uk/co/sainsburys/policies/notifications/ParameterNotificationListenerKafka.java
package uk.co.sainsburys.policies.notifications;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.kafka.annotation.KafkaHandler;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Component;

import uk.co.sainsburys.parameters.Parameter;

@Component
@KafkaListener(id = "policies", topics = "${kafka.topic.name}")
public class ParameterNotificationListenerKafka {
    private static final Logger LOGGER = LoggerFactory.getLogger(ParameterNotificationListenerKafka.class);
    @KafkaHandler
    public void parameterNotificationListenerKafka(Parameter parameter) {
        LOGGER.info("POLICIES: Parameter update via Kafka - {} = {}", parameter.getName(), parameter.getValue());
    }
}

File: spring-boot-sampler/dbaudit/src/main/kotlin/uk/co/sainsburys/dbaudit/ParameterNotificationListenerKafka.kt
package uk.co.sainsburys.dbaudit

import org.springframework.kafka.annotation.KafkaHandler
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Component

@Component
@KafkaListener(id = "dbaudit", topics = ["\${kafka.topic.name}"])
class ParameterNotificationListenerKafka(val auditRepository: AuditRepository ) {
    @KafkaHandler
    fun parameterNotificationListenerKafka(parameter: String?) {
        val auditEntity: AuditEntity = AuditEntity(null, "We modified a parameter: ${parameter}", null)
        auditRepository.save(auditEntity)
    }
}
